modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedDocument.java

```diff
@@ -56,31 +106,44 @@ public int getRating() {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(index);
+        out.writeString(type);
         out.writeString(docId);
         out.writeVInt(rating);
     }
 
-    public static RatedDocument fromXContent(XContentParser parser) throws IOException {
-        String id = null;
-        int rating = Integer.MIN_VALUE;
-        Token token;
-        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            if (parser.currentToken().equals(Token.FIELD_NAME)) {
-                if (id != null) {
-                    throw new ParsingException(parser.getTokenLocation(), "only one document id allowed, found [{}] but already got [{}]",
-                            parser.currentName(), id);
-                }
-                id = parser.currentName();
-            } else if (parser.currentToken().equals(Token.VALUE_NUMBER)) {
-                rating = parser.intValue();
-            } else {
-                throw new ParsingException(parser.getTokenLocation(), "unexpected token [{}] while parsing rated document",
-                        token);
-            }
+    public static RatedDocument fromXContent(XContentParser parser, RankEvalContext context) throws IOException {
```

Comment:
Nit: If we don't really need the context in the parser (e.g. for parsing nested queries) and this argument is only here to satisfy ObjectParse.parse(), then this could be simply ParseFieldMatcherSupplier.